extensions/hypersonic/src/features/todo-list/hooks/use-todo-list.ts

```diff
@@ -113,119 +137,242 @@ export function useTodoList() {
     }
   }
 
-  const handleSetDate = async (
-    todo: Todo,
-    dateValue: string | null,
-    name: string
-  ) => {
+  const handleSetProject = async (todo: Todo, project: any | null) => {
     try {
-      await showToast(Toast.Style.Animated, 'Scheduling...')
-
-      await mutate(updateTodoDate(todo.id, dateValue), {
+      const projectId = project?.id ? project.id : null
+      await mutate(updateTodoProject(todo.id, projectId), {
         optimisticUpdate(data) {
           if (!data) return data
-          return data.filter((t) => t.id !== todo.id)
+          return data.map((t) => (t.id === todo.id ? { ...todo, project } : t))
         },
         shouldRevalidateAfter: true,
       })
-
-      await showToast(Toast.Style.Success, `Reminder set for ${name}`)
     } catch (e: any) {
       showToast(Toast.Style.Failure, e?.message)
     }
   }
 
-  const handleDelete = async (todoId: string) => {
+  const handleSetUser = async (todo: Todo, user: any | null) => {
     try {
-      await mutate(deleteTodo(todoId), {
+      const userId = user?.id ? user.id : null
+      await mutate(updateTodoUser(todo.id, userId), {
         optimisticUpdate(data) {
           if (!data) return data
-          return data.filter((t) => t.id !== todoId)
+          return data.map((t) => (t.id === todo.id ? { ...todo, user } : t))
         },
-        shouldRevalidateAfter: false,
+        shouldRevalidateAfter: true,
       })
     } catch (e: any) {
       showToast(Toast.Style.Failure, e?.message)
     }
   }
 
-  const handleMoveUp = async (fromIndex: number) => {
+  const handleSetDate = async (
+    todo: Todo,
+    dateValue: string | null,
+    name: string
+  ) => {
     try {
-      await mutate(undefined, {
+      await showToast(Toast.Style.Animated, 'Scheduling...')
+
+      await mutate(updateTodoDate(todo.id, dateValue), {
         optimisticUpdate(data) {
-          if (!data || fromIndex === 0) return data
-          const toIndex = fromIndex - 1
-          const newTodos = [...data]
-          const todoMoved = newTodos.splice(fromIndex, 1)[0]
-          newTodos.splice(toIndex, 0, todoMoved)
-          return newTodos
+          if (!data) return data
+          return data.map((t) =>
+            t.id === todo.id
+              ? {
+                  ...todo,
+                  dateValue: dateValue,
+                  date: new Date(dateValue || ''),
+                }
+              : t
+          )
         },
-        shouldRevalidateAfter: false,
+        shouldRevalidateAfter: true,
       })
+
+      await showToast(Toast.Style.Success, `Scheduled for ${name}`)
     } catch (e: any) {
       showToast(Toast.Style.Failure, e?.message)
     }
   }
 
-  const handleMoveDown = async (fromIndex: number) => {
+  const handleDelete = async (todoId: string) => {
```

Comment:
I would add a `confirmAlert` in this function like this instead to ensure nothing wrong is deleted:

```
  const handleDelete = async (todoId: string) => {
    if (await confirmAlert({ title: 'Are you sure you want to delete?' })) {
      try {
        await mutate(deleteTodo(todoId), {
          optimisticUpdate(data) {
            if (!data) return data
            return data.filter((t) => t.id !== todoId)
          },
          shouldRevalidateAfter: false,
        })

        showToast(Toast.Style.Success, 'Task Deleted')
        refreshMenuBar()
      } catch (e: any) {
        showToast(Toast.Style.Failure, e?.message)
      }
    }
  }
```

Reply:
Cool 😎 it feels better integrated than I thought. It happened to me sometimes to delete a task by accident.

Reply:
Happens to me all the time too 😊